// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package scql.engine.auth.pb;

message PartyIdentity {
  string party_code = 1;
  // base64 encoded version of the DER encoded public key
  string public_key = 2;
}

message DriverIdentity {
  // base64 encoded version of the DER encoded public key
  string public_key = 1;
}

message AuthorizedProfile {
  DriverIdentity driver = 1;
  repeated PartyIdentity parties = 2;
}
